### PR TITLE
RIA-7200 Internal ADA hearing bundle ready letter generation

### PR DESCRIPTION
**Pull request by hussain-faruk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-7200

### Change description ###
* Added generate hearing bundle doc generation for internal ADA cases
* Added unit/functional tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/InternalAdaGenerateHearingBundleTemplateTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
+package uk.gov.hmcts.reform.iacasedocumentsapi.domain.templates;
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/InternalAdaGenerateHearingBundleTemplateTest.java:1]: Can we please use the letter package for all the notification related letter generation?

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/InternalAdaGenerateHearingBundleTemplate.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.iacasedocumentsapi.domain.templates;
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/InternalAdaGenerateHearingBundleTemplate.java:1]: Can we please use the letter package for all the notification related letter generation?

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/InternalAdaGenerateHearingBundleHandler.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.iacasedocumentsapi.domain.handlers.presubmit;
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/InternalAdaGenerateHearingBundleHandler.java:1]: Can we please use the letter package for all the notification related letter generation?

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -151,6 +151,11 @@ decisionAndReasonsFinalPdf.fileName: "Decision-and-reasons-FINAL"
 hearingBundle.fileExtension: PDF
 hearingBundle.fileName: "hearing-bundle"
 
+internalAdaHearingBundle.contentType: application/pdf
+internalAdaHearingBundle.fileExtension: PDF
+internalAdaHearingBundle.fileName: "ADA appellant hearing bundle"
+internalAdaHearingBundle.templateName: ${IA_END_APPEAL_AUTOMATICALLY_TEMPLATE:TB-IAC-DEC-ENG-00007.docx}
```
Comment (neha-aggarw) [inline src/main/resources/application.yaml:157]: can you please change the system variable name on line 157.

File: src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/InternalAdaGenerateHearingBundleHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
+package uk.gov.hmcts.reform.iacasedocumentsapi.domain.handlers.presubmit;
```
Comment (neha-aggarw) [inline src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/InternalAdaGenerateHearingBundleHandlerTest.java:1]: Can we please use the letter package for all the notification related letter generation?

**Review by neha-aggarw — CHANGES_REQUESTED** [review]

**Review by colmlynch60 — DISMISSED** [review]
LGTM

**Review by ikirsanov — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/letter/InternalAdaGenerateHearingBundleHandler.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && callback.getEvent() == Event.GENERATE_HEARING_BUNDLE
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/letter/InternalAdaGenerateHearingBundleHandler.java:46]: Same concept for this as well. We need to check the async stitching event.

**Review by neha-aggarw — CHANGES_REQUESTED** [review]

**Review by ikirsanov — APPROVED** [review]

**Review by ikirsanov — APPROVED** [review]

**Review by neha-aggarw — DISMISSED** [review]

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by ikirsanov — APPROVED** [review]

**Pull request merged**